Source/Core/Core/Movie.cpp

```diff
@@ -683,32 +685,37 @@ static void SetWiiInputDisplayString(int remoteID, const u8* const data,
       display_str += " 2";
     if (buttons.home)
       display_str += " HOME";
+  }
 
-    // A few bits of accelData are actually inside the coreData struct.
-    if (accelData)
-    {
-      wm_accel dt;
-      std::memcpy(&dt, accelData, sizeof(dt));
+  if (rpt.HasAccel())
+  {
+    DataReportBuilder::AccelData accel_data;
+    rpt.GetAccelData(&accel_data);
 
-      display_str +=
-          StringFromFormat(" ACC:%d,%d,%d", dt.x << 2 | buttons.acc_x_lsb,
-                           dt.y << 2 | buttons.acc_y_lsb << 1, dt.z << 2 | buttons.acc_z_lsb << 1);
-    }
+    // FYI: This will only print partial data for interleaved reports.
+
+    display_str += StringFromFormat(" ACC:%d,%d,%d", accel_data.x, accel_data.y, accel_data.z);
   }
 
-  if (irData)
+  if (rpt.HasIR())
   {
+    const u8* const irData = rpt.GetIRDataPtr();
+
+    // TODO: This does not handle the different IR formats.
+
     u16 x = irData[0] | ((irData[2] >> 4 & 0x3) << 8);
     u16 y = irData[1] | ((irData[2] >> 6 & 0x3) << 8);
     display_str += StringFromFormat(" IR:%d,%d", x, y);
   }
 
   // Nunchuk
-  if (extData && ext == 1)
+  if (rpt.HasExt() && ext == 1)
```

Comment:
As long as you're touching this, use the enum name too.

Reply:
I do plan on a lot of cleanups to TAS later, but I can do this real quick.